### ISSUE TITLE
Expand functionality to support "remote port forwarding" (client connection) / advanced socat / netcat functionality

### ISSUE THREAD

**Issue by brandonros** [tcp-warp/tcp-warp#18]
Machine1 with port unblocked: tcp-warp server --listen 0.0.0.0:3000

Machine2 with port block: tcp-warp client -t machine1ExternalIp:3000

Ok. They are connnected. Now what? I want to give machine1 access to machine2's port 3000.

**Comment by kilork** [comment]
Sorry, have not worked on this project for a while, did it because of my very specific case and decided to publish it for the future myself and for everyone. But till now I saw almost no interest to this project from anyone (including me, because this very specific case is now not anymore a real life case for me).

I think you found and documented solution in you PR: https://github.com/tcp-warp/tcp-warp/pull/19. It is great, thank you!

I am going to merge it and probably produce new version with fresh dependencies.

**Comment by brandonros** [comment]
Not quite. Your stuff was really close. I am looking for `ssh -R` functionality without `ssh` because `ssh` does TCP over TCP and is slow.

https://github.com/brandonros/tcpsoup

I came up with this. It allows backwards connecting to an HTTP server that is port blocked when the HTTP client is bindable.

It'd be cooler if `tcp-warp` supported this. 

The biggest problem I saw was: https://github.com/tcp-warp/tcp-warp/blob/master/tcp-warp/src/client.rs#L4

Clients can only bind.  If a client could connect optionally, I think that'd be really cool.

Take a look at `tcpsoup` and tell me if you think I could make it better or if we could combine projects. I prefer yours lol

**Comment by brandonros** [comment]
```
netcat listener1 binds on diag:0.0.0.0:5555 -> accepts connection from vehicle-tunnel-client
netcat listener2 bind on diag:127.0.0.1:3000 and pipes to vehicle-tunnel-client? (how to get access to the socket between netcat-listener1 and vehicle-tunnel-client)
```

Basically explains the problem it's in the README on TCP soup. Unless you and I both don't understand how to properly use `socat` or `netcat`.

**Comment by kilork** [comment]
Ah, okay 👍 I will take a look over weekend and keep this issue open. I discovered there are some stars and also I left a bunch of unimplemented features, so maybe it worth it.
